File: HackerRank/CompareTuples/CompareTuples.py

aliceTriplet = ()
bobTriplet = ()

def get_triplet(which):
    returnTriplet = []
    for val in range(3):
        returnTriplet.append(int(input("%s Triplet item %d: " % (which, val+1))))
    return returnTriplet

aliceTriplet = get_triplet("alice")
bobTriplet = get_triplet("bob")

def get_results(alice: list, bob: list):
    ascore, bscore = 0, 0
    for aliceVal, bobVal in zip(alice, bob):
        assert 1 <= aliceVal <= 100,\
            "Alice's answer should have been between 1 and 100, inclusive, but it was %d" % aliceVal
        assert 1 <= bobVal <= 100,\
            "Bob's answer should have been between 1 and 100, inclusive, but it was %d" % bobVal
        if aliceVal > bobVal:
            ascore+=1
        if aliceVal < bobVal:
            bscore+=1
    return ascore, bscore

aliceScore, bobScore = get_results(aliceTriplet, bobTriplet)
print(aliceScore, bobScore)

File: HackerRank/CompareTuples/CompareTuples.rb

def get_triplet(which)
    returnTriplet = []
    for val in 0..2 do
        print "#{which} triplet item ##{val+1}: "
        returnTriplet.append gets.chomp.to_i
    end
    return returnTriplet
end

aliceTriplet = get_triplet "alice"
bobTriplet = get_triplet "bob"

def get_results(alice, bob)
    ascore, bscore = 0, 0

    for aliceVal, bobVal in alice.zip(bob) do
        if (1 >= aliceVal || aliceVal >= 100) then
            puts "Alice's answer should have been between 1 and 100, inclusive, but it was #{aliceVal}"
        end
        if (1 >= bobVal || bobVal >= 100) then
            puts "Bob's answer should have been between 1 and 100, inclusive, but it was #{bobVal}"
        end
        ascore +=1 if aliceVal>bobVal
        bscore +=1 if aliceVal<bobVal
    end
    return ascore, bscore
end

aliceScore, bobScore = get_results(aliceTriplet, bobTriplet)
puts "#{aliceScore} #{bobScore}"
